shimmingtoolbox/unwrap/prelude.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+"""
+Wrapper to FSL Prelude (https://fsl.fmrib.ox.ac.uk/fsl/fslwiki/FUGUE/Guide#PRELUDE_.28phase_unwrapping.29)
+"""
+
+import numpy as np
+import os
+import nibabel as nib
+import subprocess
+
+
+def prelude(wrapped_phase, mag, affine, mask=np.array([-1]), path_2_unwrapped_phase='./unwrapped_phase.nii',
+            is_unwrapping_in_2d=True, is_saving_nii=False):
+    """wrapper to FSL prelude
+
+    This function enables phase unwrapping by calling FSL prelude on the command line. A mask can be provided to mask
+    the phase image provided. 2D unwrapping can be turned off. The output path can be specified. The temporary niis
+    can optionally be saved.
+
+    Args:
+        wrapped_phase (numpy.ndarray): 3D radian numpy array to perform phase unwrapping. (2 pi interval)
+        mag (numpy.ndarray): 3D magnitude numpy array corresponding to the phase array
+        affine (numpy.ndarray): 2D array (4x4) containing the transformation coefficients. Can be calculated by using:
+            nii = nib.load("nii_path")
+            affine = nii.affine
+        mask (numpy.ndarray, optional): numpy array of booleans with shape of `complex_array` to mask during phase
+                                        unwrapping
+        path_2_unwrapped_phase (string, optional): relative or absolute path to output the nii unwrapped phase
+        is_unwrapping_in_2d (bool, optional): prelude parameter to unwrap in 2d
+        is_saving_nii (bool, optional): specify whether `wrapped_phase`, `mag`, `affine`, `mask` and `unwrapped_phase`
+                                        nii files will be saved
+
+    Returns:
+        numpy.ndarray: 3D array with the shape of `complex_array` of the unwrapped phase output from prelude
+    """
+    # Get absolute path
+    abs_path = os.path.abspath(path_2_unwrapped_phase)
+    data_save_directory = os.path.dirname(abs_path)
+
+    # Make sure directory exists, if not create it
+    if not os.path.exists(data_save_directory):
+        print('\nCreating directory for unwrapped phase at {}'.format(data_save_directory))
```

Comment:
use logging in API

Reply:
The section of code calling print has been removed. No need to use logging in this case